PacketDotNet/IPv4Packet.cs

```diff
@@ -109,6 +109,21 @@ public override Int32 HeaderLength
             }
         }
 
+        /// <value>
+        /// The protocol of the ip packet's payload
+        /// Included along side Protocol for user convenience
+        /// 
+        /// </value>
+        [Obsolete("Use Protocol instead of NextHeader for IPv4 Packets.")]
+#pragma warning disable 0809
+        public override IPProtocolType NextHeader
```

Comment:
I think Obsolete is OK, but the same pretty much applies for IPv6's Protocol (it doesn't have Protocol in its header either).

Reply:
Yeah, correct, it's still payload protocol, it's something for the future though, it's OK right now.